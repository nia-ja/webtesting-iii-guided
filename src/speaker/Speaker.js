import React from 'react';

function Speaker(props) {
    const { speak, message="" } = props;

    return (
        <>
            <button onClick={speak}>Speak</button>
            <div>{message}</div>
        </>
    )
}

export default Speaker;