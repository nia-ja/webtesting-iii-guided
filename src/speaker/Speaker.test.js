import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/react/cleanup-after-each'; // auto clean up after each test
import "jest-dom/extend-expect";


import Speaker from './Speaker';

describe('<Speaker />', () => {
  xit('matches snapshot', () => {
    const tree = renderer.create(<Speaker />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it.skip('matches snapshot with message', () => {
    const tree = renderer.create(<Speaker message={'hello'} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should call speak on button click', () => {
    const speak = jest.fn();
    const { getByText } = render(<Speaker speak={speak} />);

    const button = getByText(/speak/i);

    fireEvent.click(button);
    fireEvent.click(button);

    expect(speak).toHaveBeenCalled();
    expect(speak).toHaveBeenCalledTimes(2);
  });
});