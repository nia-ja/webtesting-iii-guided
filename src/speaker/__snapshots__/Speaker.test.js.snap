// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Speaker /> matches snapshot 1`] = `
Array [
  <button>
    Speak
  </button>,
  <div />,
]
`;

exports[`<Speaker /> matches snapshot with message 1`] = `
Array [
  <button>
    Speak
  </button>,
  <div>
    hello
  </div>,
]
`;
